### DISCUSSION TITLE
Communicate by TP/UART

### DISCUSSION THREAD

**Discussion by svilchezl** [Supergiovane/node-red-contrib-knx-ultimate#488] [General]
Good morning,

First of all, tell you that I have been using the node for a long time, and the operation it has is incredible, congratulations, as a knx integrator for more than 20 years, the best solution that I have seen in all this time. 

This has raised me to use the node as a logic and communication engine in many current projects, and a doubt-problem has also arisen. Is it possible to use the node through a direct communication interface to knx? that is to say with a KNX / TP UART type Weinzierl BAOS 838 or similar? 
It would be a possible solution to not depend on a router ... I am thinking of a physical communication interface between knx and bacnet-modbus ...

**Comment by Supergiovane** [comment]
Hi svilchezl,
Thank you!
Natively, this is not possible and i have no plans to add this feature, as i'm focused on knx secure at this time.
But.... some other users are using knxd with success. An user has built an entire hardware native interface using knx-ultimate and knxd and is about to put it on the market.
You can do that way. 
Install knxd and simply point the knx ultimate config gateway node to 127.0.0.1 to be able to reach knxd.
Then, knxd "transforms" the IP packets in serial TP ones.
I cannot help you any further, because i don't personally use knxd, so my knowledge on that is very little.
Hope this helps.

PS: if you're a PRO, i can put your logo and webpage link, to the main README page in gitHub, if you wish. Totally free.

**Comment by mgoeller** [comment]
@Supergiovane Hi, can you give me the username of that hardware producer?

**Comment by Supergiovane** [comment]
> @Supergiovane Hi, can you give me the username of that hardware producer?

Hi mgoeller,
As far i can recall, it was [songzh96](https://github.com/songzh96)
It was maybe using a weinzierl serial to KNX interface.

**Comment by Supergiovane** [comment]
The FT1.2 Serial kberry connection is ongoing and will be released shortly

**Comment by Supergiovane** [comment]
https://youtu.be/jZ-3JYtQQ70
https://youtu.be/O6_JaNRXkwo

**Comment by Supergiovane** [comment]
https://supergiovane.github.io/node-red-contrib-knx-ultimate/wiki/-Sample---FT12

**Comment by sunshine-hass** [comment]
The Siemens TPUART module I have does not support the FT1.2 protocol functionality.
